CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+cmake_minimum_required(VERSION 3.0)
+project(YOBEMAG C)
+
+set(PRODUCT_NAME yobemag)
+string(TOUPPER ${PRODUCT_NAME} MSG_PREFIX)
+set(CMAKE_C_STANDARD 17)
+
+message("[${MSG_PREFIX}] Using ${CMAKE_GENERATOR}")
+
+if (CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "DEBUG" OR CMAKE_BUILD_TYPE STREQUAL "debug")
+    message("[${MSG_PREFIX}] Using -g")
+    add_compile_options(-g)
+endif()
+
+add_compile_options(
+        -std=c17
```

Comment:
Good question. I have no idea tbf. Apparently CMAKE_C_STANDARD isn't even defined by CMake

Reply:
Maybe just use https://cmake.org/cmake/help/latest/prop_tgt/C_STANDARD.html for that, as it seems to also set the correct flags.

Reply:
https://cmake.org/cmake/help/latest/prop_tgt/C_STANDARD.html This looks like a better option

Reply:
Nvm

> https://cmake.org/cmake/help/latest/prop_tgt/C_STANDARD.html This looks like a better option

Nvm this, I was too slow 😂

Reply:
We should also look into requiring C17 since C_STANDARD has decaying (switch to < C-17) when C17 is not available without giving errors:
https://cmake.org/cmake/help/latest/prop_tgt/C_STANDARD_REQUIRED.html#prop_tgt:C_STANDARD_REQUIRED